iris_grib/message.py

```diff
@@ -237,14 +237,29 @@ def _bitmap(self, bitmap_section):
         return bitmap
 
     def __getitem__(self, keys):
-        # NB. Currently assumes that the validity of this interpretation
+        # N.B. Assumes that the validity of this interpretation
         # is checked before this proxy is created.
 
         message = self.recreate_raw()
         sections = message.sections
+        data = None
+
+        if 5 in sections:
```

Comment:
This defensive check is necessary, as we have tests with no encoded section 5 entries in the message.

Otherwise we generate a `KeyError` when attempting an unprotected `sections[5]` look-up.